asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py

```diff
@@ -60,6 +74,29 @@ def prep(
     def provenance(self) -> dict[str, str]:
         ...
 
+    @staticmethod
+    def cache(
+        prepped_complexes: list[PreppedComplex],
+        dir: Union[str, Path],
+        type=CacheType.DesignUnit,
+    ) -> None:
+        """
+        Cache a set of design units for use later.
+        """
+        dir = Path(dir)
+        if not dir.exists():
+            dir.mkdir(parents=True)
+
+        for pc in prepped_complexes:
```

Comment:
does it make sense to first check if the file exists before saving? or is this only going to be called with newly generated structures?

Reply:
Makes sense as avoiding a general foot gun.